loki/transformations/hoist_variables.py

```diff
@@ -152,6 +152,11 @@ def transform_subroutine(self, routine, **kwargs):
         for child in successors:
             if not isinstance(child, ProcedureItem):
                 continue
+
+            # skip successors that might have been inlined
+            if not call_map.get(child.local_name, None):
```

Comment:
Shouldn't that simply be
```suggestion
            if child.local_name not in call_map:
```

Reply:
As discussed offline, this can be dropped now because `InlineTransformation.creates_items = True` if `inline_marked = True`.